### DISCUSSION TITLE
[@types/node] Export TestContext type from node:test

### DISCUSSION THREAD

**Discussion by jamiebuilds** [DefinitelyTyped/DefinitelyTyped#67127] [Issues with a @types package]
It's sometimes useful to be able to create reusable helpers that use `t: TestContext` from the `TestFn`

```ts
import test from "node:test"
import assert from "node:assert/strict"
import { groupWhile } from "./groupWhile.tsx"

// Would be better if we could import this type
type TestContext = Parameters<NonNullable<Parameters<typeof test>[0]>>[0]

function check(t: TestContext, input: number[], expected: number[][]) {
	let result = groupWhile(input, (item, index, array) => {
		return array[index - 1]! + 1 === item
	})
	t.diagnostic(`result = ${JSON.stringify(result)}`)
	t.diagnostic(`expected = ${JSON.stringify(expected)}`)
	assert.deepEqual(result, expected)
}

test("empty", (t) => {
	check(t, [], [])
})
test("sequential", (t) => {
	check(t, [1, 2, 3, 4, 5, 6], [[1, 2, 3, 4, 5, 6]])
})
test("non-sequential", (t) => {
	check(t, [1, 2, 4, 5], [[1, 2], [4, 5]])
})
```

[a comment by an automated account is not shown]

**Comment by jehon** [comment]
Same problem.
The code is located here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/test.d.ts#L391
But how do I import that into another file?

**Reply by khaosdoctor** [reply]
The `TestContext` is not exported, you would need to add `export type TestContext = TestContext` at the bottom of this declaration so you could access it through your code.

**Reply by jehon** [reply]
Thanks for guidance. I just made a PR.

**Comment by JohanAlteruna** [comment]
I'd also like to see this change; perhaps there are other types in the `node:test` module that should be exported as well. The Node documentation for version 21 [describes this class in detail](https://nodejs.org/docs/latest-v21.x/api/test.html#class-testcontext), although it does state that _"...the `TestContext` constructor is not exposed as part of the API."_. IMO, therefore it makes sense for the declarations to somehow export the type but not the class itself.

**Comment by khaosdoctor** [comment]
Same problem here, I opened another one and completely missed this: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/69488 

but still fully support this.

**Comment by jehon** [comment]
PR proposed here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69496

**Comment by khaosdoctor** [comment]
Oh darn did the same 🤣 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69497

You will need to backport this to other versions too I think, from v2x current to v18 and v16 though. Wanna switch?

**Comment by Semigradsky** [comment]
TestContext was already added, it can be closed.
